src/V3Const.cpp

```diff
@@ -590,6 +590,86 @@ class ConstVisitor : public AstNVisitor {
         return false;
     }
 
+    // 'out' will have those bits set, which are known to be set
+    void setBits(V3Number& out, AstNode* nodep) {
+        // @wsnyder Do I need some sizedness test here, or are they already well formed?
+        if (const AstConst* const np = VN_CAST_CONST(nodep, Const)) {
+            out.opAssign(np->num());
+        } else if (const AstOr* const np = VN_CAST_CONST(nodep, Or)) {
+            // @wsnyder Do I need to check operands are sized and have the same width?
+            V3Number lb(nodep, np->lhsp()->widthMin());
+            V3Number rb(nodep, np->rhsp()->widthMin());
+            setBits(lb, np->lhsp());
+            setBits(rb, np->rhsp());
```

Comment:
I believe this recursion makes O(n^2) behavior in large and/or trees which will be a problem.  Maybe limit recursion depth?

Reply:
It's only invoked on the operands of AstEq and AstNeq at the moment, so should only visit every node in those at most once. They also only recurse through pure AstOr trees, not any other nodes. I will add a comment about this but otherwise I think it's OK as is right now, and can revisit if more nodes are handled ever.